06-classification_continued.Rmd

```diff
@@ -236,9 +321,12 @@ different train / validation splits, we'll get a better estimate of accuracy,
 which will lead to a better choice of the number of neighbours $K$ for the
 overall set of training data. 
 
-> **Note:** you might be wondering why we can't we use the multiple splits to test our final classifier after tuning is done. This is simply
-> because at the end of the day, we will produce a single classifier using our overall training data. If we do multiple train / test splits, we will
-> end up with multiple classifiers, each with their own accuracy evaluated on different test data.
+> **Note:** you might be wondering why we can't we use the multiple splits to
+> test our final classifier after tuning is done. This is simply because at the
+> end of the day, we will produce a single classifier using our overall
+> training data. If we do multiple train / test splits, we will end up with
+> multiple classifiers, each with their own accuracy evaluated on different
+> test data.
 
 Let's investigate this idea in R! In particular, we will use different seed
 values in the `set.seed` function to generate five different train / validation
```

Comment:
Our learners have never seen for loops before... This is extra cognitive load this will distract from our point. Our point is that we want to illustrate that different splits will give us different accuracy estimates. I wonder if a simpler way to illustrate this, with less cognitive load, would be to use one seed to calculate one accuracy with code and show it and discuss it in the text. Then in another code chunk, repeat the code (yes it's not DRY code, but that is not the point here) using a different seed and show we get a different accuracy. Then in the text, say if we did this this 3 more times with different seeds (we could even list them in the text) we would get accuracies X, Y and Z (we do the calculation in a hidden code cell). This would avoid the loops, which they don't know, but still get our point across.
If you really want to use a loop here, then I think we have to teach loops in the wrangling chapter.

Also, related to my comment above, I think we should show this under a heading called "Validation accuracy" and then this leads into can we do something better? And yes we can, the better thing is cross validation and we can then explain that. So just moving the CV text above below this.

Reply:
removed for loop